stubs/psycopg2/psycopg2/_psycopg.pyi

```diff
@@ -342,14 +342,14 @@ class connection:
     deferrable: Any
     dsn: Any
     encoding: Any
-    info: Any
+    info: ConnectionInfo
     isolation_level: Any
     notices: Any
     notifies: Any
     pgconn_ptr: Any
-    protocol_version: Any
+    protocol_version: int
     readonly: Any
-    server_version: Any
+    server_version: int
```

Comment:
Does that include notices, notifies and cursor_factory? They aren't marked with `READONLY` in that array of structs.

The psycopg2 docs say that [notices and notifies are user-writable](https://www.psycopg.org/docs/connection.html#connection.notices) since psycopg 2.7. No clues on cursor_factory though.